analyses/rnaseq-batch-correct/code/02-ruvseq-deseq-tn.R

```diff
@@ -0,0 +1,385 @@
+design <- model.matrix( ~ DESeqGroup)
+counts_object <- edgeR::DGEList(counts = cnt_df, group = DESeqGroup)
+counts_object_filtered <-
+  edgeR::filterByExpr(counts_object, group = DESeqGroup)
+counts_object <-
+  counts_object[counts_object_filtered, , keep.lib.sizes = FALSE]
+
+# create new expression set
+seq_expr_set <-
+  EDASeq::newSeqExpressionSet(
+    counts = round(counts_object$counts),
+    phenoData = data.frame(bs_id, DESeqGroup, RNA_library, row.names = colnames(round(
+      counts_object$counts
+    )))
+  )
+
+########################################### DESeq2 analysis ##############################################################
+
+# 1.1. RUVg using hk genes in normals only (full HRT atlas)
+message("Running RUVg-DESeq2 analysis")
+neg_ctrl_genes <- readRDS(file.path(input_dir, neg_ctrl_genes.f))
+ruvg_out <-
+  ruvg_test(
+  seq_expr_set = seq_expr_set,
+  k_val = c((drop + 1):k_value),
+  emp_neg_ctrl_genes = neg_ctrl_genes,
+  prefix = "hk_genes_normals",
+  diff_type = "deseq2",
+  output_dir = output_dir,
+  plot_dir = plots_dir,
+  design_variable = "DESeqGroup",
+  color_var = "DESeqGroup",
+  shape_var = "RNA_library",
+  drop = drop
+)
+
+message("Batch correction DGE complete")
+
+#  Non-batch corrected DESeq2::DESeq performs a default analysis through the steps:
+# - estimation of size factors: estimateSizeFactors
+# - estimation of dispersion: estimateDispersions
+# - Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest
+
+message("Running non-batch corrected DESeq2 analysis")
+dds_nobatch <-
+  DESeq2::DESeqDataSetFromMatrix(
+    countData = round(counts_object$counts),
+    colData =  data.frame(bs_id, DESeqGroup),
+    design = design
+  )
+dds_nobatch <- DESeq2::DESeq(dds_nobatch)
+dge_output_nobatch <-
+  DESeq2::results(dds_nobatch,
+                  cooksCutoff = FALSE,
+                  pAdjustMethod = 'BH')
+dge_output_nobatch <- dge_output_nobatch %>%
+  as.data.frame() %>%
+  rownames_to_column('gene') %>%
+  arrange(padj)
+
+message("Writing DESeq2 non batch-corrected results")
+readr::write_tsv(x = dge_output_nobatch,
+                 file = file.path(
+                   output_dir,
+                   'deseq2_analysis',
+                   paste0('DESeq2_noBatchC_', dataset, '_dge.tsv')
+                 ))
+
+# plot and save p-value histogram
+message("Saving p-value histogram")
+p <- deseq2_pvals_histogram(
+  res_df = dge_output_nobatch,
+  xlab = 'DGE RLE nbinomWaldTest p-value',
+  ylab = 'Gene count',
+  title = paste0('Histogram of DESeq2 nbinomWaldTest p-values')
+)
+filename <- file.path(plots_dir, 'dge_deseq2_histogram.pdf')
+ggsave(
+  filename = filename,
+  plot = p,
+  width = 8,
+  height = 7,
+  bg = "white"
+)
+
+print("DESeq2 analysis finished.")
+
+# Read in positive and negative control gene sets
+pos_ctrl_genes <- readRDS(file.path(input_dir, pos_ctrl_genes.f))
+gene.list <- c(neg_ctrl_genes, pos_ctrl_genes)
```

Comment:
```suggestion
pos_neg_ctrl_genes <- c(neg_ctrl_genes, pos_ctrl_genes)
```
Suggest to make this more informative